motoman_msgs/srv/SelectTool.srv

```diff
@@ -0,0 +1,92 @@
+
+# Changes the tool file used by the controller for (Moto)ROS-controlled motions.
+#
+# NOTE 1: this changes the active tool for (Moto)ROS-controlled motions ONLY.
+#
+#         Regular INFORM programs, jogging or any other (programmed) motions
+#         executed by the controller will NOT use the new tool.
+#
+# NOTE 2: any tool switch initiated by invoking this service WILL however be
+#         taken into account by safety systems such as the FSU and PFL while
+#         executing (Moto)ROS-initiated motions.
+#
+# NOTE 3: active tool will be switched AFTER the controller has completed
+#         execution of the last trajectory segment that was sent to the
+#         controller BEFORE this service was called.
+#
+#         The motion queue internal to MotoROS caches a number of segments in
+#         a FIFO. Only segments received AFTER this service was invoked will
+#         be executed with the new tool. Any segments received before a tool
+#         switch was invoked will use the old tool.
+#
+#         To guarantee motion will be executed using the new tool, applications
+#         should check the 'in_motion' field (part of the RobotStatus message
+#         published on the 'robot_status' topic by the driver) and invoke the
+#         service when the robot has come to a stop (and the motion queue of
+#         MotoROS is empty). Any subsequent trajectories will use the new tool.
+#
+# NOTE 4: as MotoROS currently only executes joint space trajectories, changing
+#         tool file with this service DOES NOT affect TCP poses while executing
+#         trajectories.
```

Comment:
But the TCP will be honored for things such as safety zones.

Reply:
Yes. I hoped `NOTE 2` would already clarify this?

Reply:
I think it is confusing to refer the "TCP poses" here.  How about simply say:  "...DOES NOT affect the robot trajectory."
We could also add:  "However, proper tool selection is important as it is used for motion dynamics and safety calculations."

Reply:
I've used slightly more generic language to clarify this.

Please check whether the current phrasing is OK.